analysers/analyser_osmosis_powerline.py

```diff
@@ -24,28 +24,110 @@
 from .Analyser_Osmosis import Analyser_Osmosis
 from modules.Stablehash import stablehash64
 
-# Power lines nodes with their voltage as array padded up to 99 zeros
+# Power lines nodes with their voltage as array padded up to 99 zeros to cope with non-numerical values
 # Lines with no voltage get null voltage instead empty array
 sql01 = """
 CREATE TEMP TABLE power_lines_nodes AS
 SELECT
-    ways.id as wid,
-    unnest(ways.nodes) AS nid,
-    unnest(ways.nodes[2:array_length(ways.nodes,1)]) AS nid_next,
-    ways.tags->'cables' as cables,
-    ways.tags->'circuits' as circuits,
-    coalesce(ways.tags->'location','overhead') as location,
+    w.id as wid,
+    unnest('{NULL}' || w.nodes[1:array_length(w.nodes, 1) - 1]) AS nid_prec,
+    unnest(w.nodes) AS nid,
+    unnest(w.nodes[2:array_length(w.nodes, 1)]) AS nid_next,
+    w.tags->'cables' AS cables,
+    coalesce((w.tags->'circuits')::integer, 1) AS circuits,
+    coalesce(w.tags->'location', 'overhead') AS location,
     voltage
 FROM
-    ways
-    JOIN LATERAL (SELECT array_agg(lpad(v, 99, '0')) FROM unnest(regexp_split_to_array(ways.tags->'voltage','; *')) AS t(v)) AS t(voltage) ON TRUE
+    ways AS w
+    JOIN LATERAL (
+        SELECT array_agg(lpad(v, 99, '0'))
+        FROM unnest(array_cat(
+            array_fill(
+                substring(w.tags->'voltage' for greatest(char_length (w.tags->'voltage'), position(';' in w.tags->'voltage')))::text, -- voltage1 in voltage1;voltage2
+                ARRAY[coalesce((w.tags->'circuits')::integer, 1) - 1 + char_length(coalesce(w.tags->'voltage', '')) - char_length(replace(coalesce(w.tags->'voltage',''), ';', ''))]
+            ),
+            regexp_split_to_array(w.tags->'voltage', '; *'))
+        ) AS t(v)) AS t(voltage)
+        ON TRUE
 WHERE
-    ways.tags != ''::hstore AND
-    ways.tags?'power' AND
-    ways.tags->'power' IN ('line', 'minor_line', 'cable')
+    w.tags != ''::hstore AND
+    w.tags?'power' AND
+    w.tags->'power' IN ('line', 'minor_line', 'cable') AND
+    w.tags->'voltage' IS NOT NULL
+
+UNION ALL
+
+SELECT
+    w.id AS wid,
+    unnest('{NULL}' || w.nodes[1:array_length(w.nodes, 1) - 1]) AS nid_prec,
+    unnest(w.nodes) AS nid,
+    unnest(w.nodes[2:array_length(w.nodes, 1)]) AS nid_next,
+    w.tags->'cables' AS cables,
+    coalesce((w.tags->'circuits')::integer, 1) AS circuits,
+    coalesce(w.tags->'location', 'overhead') AS location,
+    NULL AS voltage
+FROM
+   ways AS w
```

Comment:
There is only one table no need for alias

Reply:
It was easier to investigate bugs, just change one thing no need to change `ways` everywhere. `ways` can also be `test.ways` or `france_rhone_alpes_ain`...